src/wallet/wallet.cpp

```diff
@@ -2739,8 +2747,16 @@ bool CWallet::CreateTransaction(const std::vector<CRecipient>& vecSend, CWalletT
                     return false;
                 }
 
-                LearnRelatedScripts(vchPubKey, g_change_type);
-                scriptChange = GetScriptForDestination(GetDestinationForKey(vchPubKey, g_change_type));
+                // If -changetype is not specified and g_address_type is not legacy,
+                // and any destination is (P2SH-)P2WPKH, use P2WPKH for the change
+                // output.
+                OutputType change_type = g_change_type ? g_change_type : g_address_type;
+                if (g_change_type == OUTPUT_TYPE_NONE && g_address_type != OUTPUT_TYPE_LEGACY && any_destination_native_segwit) {
```

Comment:
This test will always fail when `g_address_type` is set to something else than `OUTPUT_TYPE_NONE`, no?

Reply:
Nevermind, I'm confusing `change_type` with `g_change_type`.